### PR TITLE
Average history graph

### LINKED ISSUE #131: Average history is drawn into a chart in algorithm details

### PR DESCRIPTION
**Pull request by VirtualAkseli**
## Summary
The graph for average history is drawn similarly to metrics. I also changed the method that saves average history to database: Multiple runs on the same metric & molecule does not update average history values. Closes #131  .

The graph's dimensions are quite big. One can zoom closer to see differences better.

## How to test
- [ ] Check that the label's are logical 
- [ ]  Ensure that benchmarking with different combinations does not cause false results.

### CODE REVIEW

File: WebCLI/views/worker_api.py [hunk 1]
```diff
@@ -15,9 +15,39 @@ def as_analyzed_results(result):
...
+    if len(avg_existing_history) < 1:
+        for i in range(len(avg_histories)):
+            history = Average_history(
+                analyzed_results=Metrics.objects.get(pk=result["metrics_id"]),
+                data=avg_histories[i],
+                iteration_number=i+1)
+            history.save()
+    else:
+        history = avg_existing_history[0]
+    if len(acc_existing_history) < 1:
```
[the remark this answers is not in the document]
Comment (VirtualAkseli) [reply WebCLI/views/worker_api.py:38]: Going around this duplication seems complex just to satisfy codeclimate.

File: WebCLI/views/worker_api.py [hunk 2]
```diff
@@ -15,9 +15,39 @@ def as_analyzed_results(result):
     return metrics
 
 
+def as_history(result):
+    avg_histories = result["average_history"]
+    avg_existing_history = Average_history.objects.filter(analyzed_results=result["metrics_id"])
+    acc_histories = result["accuracy_history"]
+    acc_existing_history = Accuracy_history.objects.filter(analyzed_results=result["metrics_id"])
+
+    if len(avg_existing_history) < 1:
```
[the remark this answers is not in the document]
Comment (VirtualAkseli) [reply WebCLI/views/worker_api.py:30]: Going around this duplication seems complex just to satisfy codeclimate.

File: WebCLI/views/worker_api.py [hunk 3]
```diff
@@ -15,9 +15,39 @@ def as_analyzed_results(result):
     return metrics
 
 
+def as_history(result):
```
[the remark this answers is not in the document]
Comment (VirtualAkseli) [reply WebCLI/views/worker_api.py:21]: This can be averted with using separate classes.

**Review by lambdakilo — APPROVED** [review]
Code seems legit and clean. With zooming the Average results are seemingly graphed out correctly.

**Pull request merged**